src/community/oseo/oseo-core/src/main/java/org/geoserver/opensearch/eo/store/JDBCCollectionFeatureStore.java

```diff
@@ -18,18 +22,40 @@
 public class JDBCCollectionFeatureStore extends AbstractMappingStore {
 
     static final Logger LOGGER = Logging.getLogger(JDBCCollectionFeatureStore.class);
+    protected String workspace;
+    static final FilterFactory2 FF = CommonFactoryFinder.getFilterFactory2();
 
     public JDBCCollectionFeatureStore(
             JDBCOpenSearchAccess openSearchAccess, FeatureType collectionFeatureType)
             throws IOException {
         super(openSearchAccess, collectionFeatureType);
     }
 
+    /**
+     * Constructor with workspace
+     *
+     * @param openSearchAccess OpenSearchAccess
+     * @param collectionFeatureType FeatureType of collection
+     * @param workspace Workspace name
+     * @throws IOException Exception if error connecting to source
+     */
+    public JDBCCollectionFeatureStore(
+            JDBCOpenSearchAccess openSearchAccess,
+            FeatureType collectionFeatureType,
+            String workspace)
+            throws IOException {
+        super(openSearchAccess, collectionFeatureType);
+        this.workspace = workspace;
+    }
+
     @Override
     protected SimpleFeatureSource getDelegateCollectionSource() throws IOException {
-        return openSearchAccess
-                .getDelegateStore()
-                .getFeatureSource(JDBCOpenSearchAccess.COLLECTION);
+        WorkspaceInfo workspaceInfo = LocalWorkspace.get();
```

Comment:
I was wondering, if there is no local workspace, should be wrap? Self answer: yes, because we want to report only collections authorized for the global workspace.